xla/service/gpu/double_buffer_loop_unrolling_test.cc

```diff
@@ -812,6 +813,457 @@ ENTRY main {
   EXPECT_EQ(num_whiles, 12);
 }
 
+TEST_F(GpuLoopDoubleBufferTransformerTest, WhileLoopWithCollectivePermute) {
+  const char* kModuleString = R"(
+HloModule loop_unrolling_no_deps
+condition {
+  input_tuple = (f32[], s32[]) parameter(0)
+  cond = s32[] get-tuple-element(input_tuple), index=1
+  trip_count = s32[] constant(10)
+  ROOT done = pred[] compare(cond, trip_count), direction=LT
+}
+ar_add {
+  Arg_1 = f32[] parameter(1)
+  Arg_0 = f32[] parameter(0)
+  ROOT add_ar = f32[] add(Arg_1, Arg_0)
+}
+body {
+  input_tuple = (f32[], s32[]) parameter(0)
+  param_0 = f32[] get-tuple-element(input_tuple), index=0
+  cond = s32[] get-tuple-element(input_tuple), index=1
+  collective-permute = f32[] collective-permute(param_0), channel_id=1, source_target_pairs={{0,1},{1,2},{2,3},{3,0}},
+                             frontend_attributes={_xla_send_recv_validation="{{0,6},{1,7},{2,8},{3,9}}"}
+  one = s32[] constant(1)
+  cond_plus_1 = s32[] add(cond, one)
+  ROOT output_tuple = (f32[], s32[]) tuple(collective-permute, cond_plus_1)
+}
+ENTRY main {
+  param_0 = f32[] parameter(0)
+  param_2 = s32[] constant(0)
+  tuple = (f32[], s32[]) tuple(param_0, param_2)
+  ROOT while = (f32[], s32[]) while(tuple), condition=condition, body=body, backend_config={"known_trip_count":{"n":"10"}}
+}
+  )";
+
+  TF_ASSERT_OK_AND_ASSIGN(std::unique_ptr<xla::HloModule> module,
+                          ParseAndReturnVerifiedModule(kModuleString));
+  DoubleBufferLoopUnrolling double_buffer(
+      DoubleBufferLoopUnrolling::UnrollStrategy::kDoubleBuffer);
+  EXPECT_THAT(double_buffer.Run(module.get()), IsOkAndHolds(true));
+
+  // Get the collective permute instructions
+  HloComputation* while_body =
+      module->entry_computation()->root_instruction()->while_body();
+  std::vector<HloInstruction*> collective_permutes;
+  for (HloInstruction* inst : while_body->instructions()) {
+    if (inst->opcode() != HloOpcode::kCollectivePermute) continue;
+    collective_permutes.push_back(inst);
+  }
+  ASSERT_EQ(collective_permutes.size(), 2);
+  ASSERT_TRUE(collective_permutes[0]->frontend_attributes().map().contains(
+      kSendRecvValidationAttr));
+  ASSERT_TRUE(collective_permutes[1]->frontend_attributes().map().contains(
+      kSendRecvValidationAttr));
+  EXPECT_EQ(collective_permutes[0]->frontend_attributes().map().at(
+                kSendRecvValidationAttr),
+            "{{0,3},{1,3},{1,4},{2,4}}");
+  EXPECT_EQ(collective_permutes[1]->frontend_attributes().map().at(
```

Comment:
Nit: maybe do a FileCheck test to give an easier at-a-glance representation of what happens to IR?